### ISSUE TITLE
unable to allocate memory from operating system at doCallback

### ISSUE THREAD

**Issue by Erick-Zanetti** [hgourvest/node-firebird#137]
I make a series of selects in my bank and I have the following error, would you have any configuration to solve this error?

`    exports.findInfoExtra = function (id, cb) {
        
        Firebird.attach(firebirdConfig, function (err, db) {
            if (err)
                throw err;

            db.query("select PRO.CODIGO, (select first 1 CODIGO from CAD_PROSPECTOS_PROPOSTAS where PROSPECTO = PRO.CODIGO order by CODIGO desc) as COD_PROPOSTA, (select first 1 CODIGO from CAD_PROSPECTOS_AGENDAMENTOS where PROSPECTO = PRO.CODIGO order by CODIGO desc) as COD_AGENDAMENTO, (select first 1 CODIGO from CAD_PROSPECTOS_CONTRATOS where PROSPECTO = PRO.CODIGO order by CODIGO desc) as COD_CONTRATO, (select first 1 CONTRATO_CONFIRMADO from CAD_PROSPECTOS_CONTRATOS where PROSPECTO = PRO.CODIGO order by CODIGO desc) as CONTRATO_CONFIRMADO from CAD_PROSPECTOS PRO where PRO.CODIGO = ?", [id], function (err, result) {
                if(result){
                    cb(result[0])
                } else{
                    cb(null)
                }
            });
        });
    }`
this function is triggered by a forEach can this influence the error?

**Comment by ericlovescode** [comment]
Don't forget you need to db.detach() on each iteration.

**Comment by Erick-Zanetti** [comment]
This will solve my memory problem so I thought this command was optional, thank you very much sorry for the inconvenience
